core/transport/messages.py

```diff
@@ -0,0 +1,99 @@
+from typing import TypeVar, Any
+
+
+class MessageBase:
+    @classmethod
+    def from_json(cls, message_json):
+        message_json.pop('msg_type')
+        return cls(**message_json)
+
+    def to_json(self) -> dict:
+        return self.__dict__
+
+
+TMessageBase = TypeVar('TMessageBase', bound=MessageBase)
```

Comment:
it's better to put all types to the bottom of the file

Reply:
But it is used in type annotation further in this file.